### ISSUE TITLE
GNOME Software is broken on master

### ISSUE THREAD

**Issue by necessarily-equal** [NixOS/nixpkgs#160530]
### Steps To Reproduce
```console
$ nix run nixpkgs#gnome.gnome-software
00:26:08:0195 GLib-GIO g_app_info_get_name: assertion 'G_IS_APP_INFO (appinfo)' failed
```
This is the default nixpkgs flake, targetting NixOS master. `nix` comes from NixOS 21.11's `pkgs.nixFlakes`

### Expected behavior
It runs

### Additional context
Host is running NixOS 21.11. GNOME Software from that version works

### Notify maintainers
@hedning @jtojnar @dasj19 @maxeaubrey

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.96, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(root): `"nixos-21.11.335820.521e4d7d13b, nixpkgs-git"`
 - channels(afontain): `"nixos-20.09-20.09.4407.1c1f5649bb9, nixpkgs-unstable-22.05pre342949.28d58b97925, nixpkgs-20.09-20.09.4407.1c1f5649bb9"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by necessarily-equal** [comment]
Also, I'm using GNOME 40 Wayland, and
```
nix-repl> :p outputs.legacyPackages.x86_64-linux.gnome.gnome-software.version
"41.3"
```

**Comment by andersk** [comment]
GNOME Software 41 from `master` works fine on GNOME 41 from `master` (or `nixos-unstable`). Given how tightly integrated GNOME is, I don’t find it surprising that you need the same version of everything. Why do you want to mix-and-match and why do you expect that to work?

**Comment by necessarily-equal** [comment]
Turns out that it actually works, but due to the gnome-software daemon being autostarted, it did unexpected things. If you close it manually then it works perfectly. The error ("assertion failed") did look like a fatal bug, and one that isn't supposed to happen, even with mix-and-match. So I opened an issue.

> why do you want to mix-and-match

This was for experimenting with a newer version to see if a bug has been fixed.

> why do you expect that to work?

Because why not! While GNOME Settings is pretty dependent on the version of the many gnome settings daemons, I wouldn't expect GNOME Software to be. I think I've even seen GNOME Software working on non-GNOME?

My mistake though, as it actually works, closing.
